Aplib.Core/Desire/IGoalPredicate.cs

```diff
@@ -0,0 +1,15 @@
+namespace Aplib.Core.Desire
+{
+    /// <summary>
+    /// This is called a predicate, but really it represents the way to calculate the current state's heuristic values.
```

Comment:
If the name isn't representative of what it does can't we call it something like "IHeuristicState" or "IHeuristicProgress" instead?

Reply:
I agree, the code comment suggests there should be a name change.

Reply:
Let's discuss it in a small meeting.